arch/arm/src/s32k1xx/s32k1xx_lpspi.c

```diff
@@ -1294,42 +1387,130 @@ static void s32k1xx_lpspi_exchange_nodma(FAR struct spi_dev_s *dev,
 {
   FAR struct s32k1xx_lpspidev_s *priv = (FAR struct s32k1xx_lpspidev_s *)dev;
   DEBUGASSERT(priv && priv->spibase);
+  uint16_t framesize;
```

Comment:
Coding standard:The variables need to be declared prior to any code execution.  Move this line above the DEBUGASSERT